code/modules/fabrication/fabricator_ui.dm

```diff
@@ -1,94 +1,211 @@
-			material_data["eject_label"] = ispath(material, /decl/material) ? "Eject" : "Flush"
-			data["material_storage"] +=    list(material_data)
-
-		var/list/current_build = list()
-		data["current_build"] = current_build
-		if(currently_building)
-			current_build["name"] =       currently_building.target_recipe.name
-			current_build["multiplier"] = currently_building.multiplier
-			current_build["progress"] =   "[100-round((currently_building.remaining_time/currently_building.target_recipe.build_time)*100)]%"
-		else
-			current_build["name"] =       "Nothing."
-			current_build["multiplier"] = "-"
-			current_build["progress"] =   "-"
-
-		data["build_queue"] = list()
-		if(length(queued_orders))
-			for(var/datum/fabricator_build_order/order in queued_orders)
-				var/list/order_data = list()
-				order_data["name"] = order.target_recipe.name
-				order_data["multiplier"] = order.multiplier
-				order_data["reference"] = "\ref[order]"
-				data["build_queue"] += list(order_data)
-		else
-			var/list/order_data = list()
-			order_data["name"] = "Nothing."
-			order_data["multiplier"] = "-"
-			data["build_queue"] += list(order_data)
-
-		data["build_options"] = list()
-		for(var/datum/fabricator_recipe/R in design_cache)
-			if(R.hidden && !(fab_status_flags & FAB_HACKED))
-				continue
-			if(show_category != "All" && show_category != R.category)
-				continue
-			if(filter_string && !findtextEx_char(lowertext(R.name), lowertext(filter_string)))
-				continue
-			var/list/build_option = list()
-			var/max_sheets = 0
-			build_option["name"] =      R.name
-			build_option["reference"] = "\ref[R]"
-			build_option["illegal"] =   R.hidden
-			if(!length(R.resources))
-				build_option["cost"] = "No resources required."
-				max_sheets = 100
-			else
-				//Make sure it's buildable and list required resources.
-				var/list/material_components = list()
-				for(var/material in R.resources)
-					var/sheets = round(stored_material[material]/round(R.resources[material]*mat_efficiency))
-					if(isnull(max_sheets) || max_sheets < sheets)
-						max_sheets = sheets
-					if(stored_material[material] < round(R.resources[material]*mat_efficiency))
-						build_option["unavailable"] = 1
-					material_components += "[round(R.resources[material] * mat_efficiency)][SHEET_UNIT] [stored_substances_to_names[material]]"
-				build_option["cost"] = "[capitalize(jointext(material_components, ", "))]."
-			if(R.max_amount >= PRINT_MULTIPLIER_DIVISOR && max_sheets >= PRINT_MULTIPLIER_DIVISOR)
-				build_option["multiplier"] = list()
-				for(var/i = 1 to FLOOR(min(R.max_amount, max_sheets)/PRINT_MULTIPLIER_DIVISOR))
-					var/mult = i * PRINT_MULTIPLIER_DIVISOR
-					build_option["multiplier"] += list(list("label" = "x[mult]", "multiplier" = mult))
-			data["build_options"] += list(build_option)
+	return data
+
+/obj/machinery/fabricator/proc/ui_data_resources(mob/user, ui_key)
+	if(!is_functioning())
+		return
```

Comment:
Do we really need to check this every time? Are these actually ever called individually in a way where the check matters?

Reply:
That's a good point. I can probably just remove most of those checks.

Reply:
fixed